records/generator.go

```diff
@@ -286,15 +287,19 @@ func (rg *RecordGenerator) masterRecord(domain string, masters []string, leader
 	h := strings.Split(leader, "@")
```

Comment:
The current func implementation makes an assumption about the order of `masters`: it's the order in which you expect the ernumerated `masterX` records to be created. I think this is probably important: if a new leader is elected, you may not want it to become `master0` simply because it's the leader. You probably want your DNS records to change as little as possible. And this func should have the least impact on enumeration order, or name/IP mappings - it's just creating the records. So let the caller do the work of ordering, sorting, whatever.. the `masters` list if a different outcome is desired.

Another consequence of the current overall mesos-dns app implementation is that the leader may not even be in the `masters` list at some point in time. `masters` is really *fallback-masters* (only consider these to be masters if I can't find a leader via ZK). At some point in time, they may not actually be masters any more. Consider a cluster of 3 nodes that suffers the loss of a member, and gains a a new member (VM crashed, was replaced by another VM). And the cycle repeats several times. You end up with a set of running masters (and leader) that's different than the set of statically configured fallback masters.

So the func tries to index the masters as they're listed and begrudgingly assigns the leading master an index out-of-band if it's not actually listed in the masters list. There are probably better ways to do it. But I'm really not trying to optimize that here. I'm just trying to fix #165 without introducing additional inconsistency into the system.

I should probably document some of this in the Go comments of the func.